From 51239e52166ab07725832f12efda367299e11cc8 Mon Sep 17 00:00:00 2001
From: Zhang Xiao <xiao.zhang@windriver.com>
Date: Tue, 12 Jun 2012 13:48:37 +0800
Subject: [PATCH] lsi_app3300: SMP conflict with PREEMPT

ARMV6 MPCore has a limitation that cache clean operation doesn't affect on other
cores. On software side, related cache handlers haven't work around it thus all
related cache flush operations on this type of CPU just local effective. On SMP
target, with preempt enabled, such limitation may cause coherent issue in text
section load & execute. Modify Kconfig to avoid such conflict.

Signed-off-by: Zhang Xiao <xiao.zhang@windriver.com>
---
 arch/arm/Kconfig          |    2 +-
 arch/arm/mach-app/Kconfig |    1 -
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index e4633ce..afa04d8 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -1163,7 +1163,7 @@ config SMP
 	bool "Symmetric Multi-Processing (EXPERIMENTAL)"
 	depends on EXPERIMENTAL && (REALVIEW_EB_ARM11MP || REALVIEW_EB_A9MP ||\
 		 MACH_REALVIEW_PB11MP || MACH_REALVIEW_PBX || ARCH_OMAP4 || ARCH_U8500 ||\
-		ARCH_APP3K)
+		 (ARCH_APP3K && PREEMPT_NONE))
 	depends on GENERIC_CLOCKEVENTS
 	select USE_GENERIC_SMP_HELPERS
 	select HAVE_ARM_SCU if (ARCH_REALVIEW || ARCH_OMAP4 || ARCH_U8500 || ARCH_APP3K)
diff --git a/arch/arm/mach-app/Kconfig b/arch/arm/mach-app/Kconfig
index a288a9a..c958f73 100644
--- a/arch/arm/mach-app/Kconfig
+++ b/arch/arm/mach-app/Kconfig
@@ -19,7 +19,6 @@ endchoice
 config ARCH_APP3K
 	bool "APP3300 family support"
 	depends on ARCH_APPXK
-	select SMP
 
 config MACH_APP33OO_MSBG
 	bool "APP3300 MSBG board"
-- 
1.7.9.7

